### PR TITLE
Use cachix authtoken in CI

### PR DESCRIPTION
**Pull request by monacoremo**

### CODE REVIEW

**Comment by monacoremo** [comment]
Will need to merge in order to see whether it works.

File: .circleci/config.yml [hunk 1]
```diff
@@ -140,9 +140,16 @@ jobs:
...
+            if [ -n "${CACHIX_AUTH_TOKEN:-""}" ]; then
               echo "Building and caching all derivations..."
+              cachix authtoken "$CACHIX_AUTH_TOKEN"
+
+              # Push new builds as we go
               nix-build | cachix push postgrest
+
+              # Make sure that everything, including .drv files, is pushed
+              nix-env -f default.nix -iA devtools
+              postgrest-push-cachix
```
Comment (wolfgangwalther) [inline .circleci/config.yml:152]: Does it need to call cachix twice?

Comment (monacoremo) [reply .circleci/config.yml:152]: To be both thorough and fast yes - the first one pushes to cachix while we build. That misses some .drv files etc though, so we need the second pass. We could only have the second, but that would mean we waste upload bandwidth while we build

Comment (wolfgangwalther) [reply .circleci/config.yml:152]: Ah, I see. Cool!

**Pull request merged**